src/useComputedStyle/useComputedStyle.ts

```diff
@@ -2,29 +2,33 @@
 import { reactive, ref, Ref, ToRefs, toRefs, watch } from 'vue'
 import { useMutationObserver } from '../useMutationObserver/useMutationObserver'
 import { useResizeObserver } from '../useResizeObserver/useResizeObserver'
-import { globalExists } from '../utilities/global'
+import { DocumentDoesNotExist } from './documentDoesNotExist'
+import { globalExists } from '@/utilities/global'
+import { ComputedStyleRecord, getComputedStyleRecord } from '@/utilities/window'
 
-export function useComputedStyle(element: Element | Ref<Element | undefined>): ToRefs<Partial<CSSStyleDeclaration>> {
+export function useComputedStyle(element: Element | Ref<Element | undefined>): ToRefs<ComputedStyleRecord> {
   const elementRef = ref(element)
-  const style = reactive({})
+  const style = reactive(getComputedStyleRecord(elementRef.value ?? getDefaultElement())!)
```

Comment:
I did this so that we can basically new up a `CSSStyleDeclaration` so our return type is consistent. Otherwise I'd have to return a partial or undefined

Reply:
We haven't done this before. But what if we overloaded the composition so if you give us a ref we'll return the declaration or null?

Reply:
I agree that this is a bit unique. Unfortunately I think this is an unintended consequence of the decision to return `toRefs` of a `reactive`. If my reactive starts as `{}`, and thus the return value is 

```ts
const { columnGap } = useComputedStyle(element)

// typeof columnGap === undefined <- not good
// different from typeof columnGap.value === undefined <- would probably work fine
```

then, even after columnGap theoretically gets populated, it's not reactive like it would be if only `.value` was undefined

Reply:
Ah right. Well I'm not a fan of the span thing but I also don't have any better ideas.

Reply:
I could move away from `toRefs` and just return `Ref<CSSStyleDeclaration | undefined>`? That would mean we can't do 

```ts
const { columnGap, paddingRight } = useComputedStyle(element)
```

instead we would have to do 

```ts
const styles = useComputedStyle(element)
// styles.value?.columnGap
```

Reply:
3rd option, define my own class that looks like `CSSStyleDeclaration` with properties like `width`, `paddingRight`, etc, but can be newed up without being given an element. One positive outcome of that approach is we could type the things that should be numbers as number instead of all being strings.

what's your favorite?

Reply:
here's what that class looks like with only the properties I need. There are approx 600 total properties on `CSSStyleDeclaration` 😬 

```ts
export class ComputedStyleRecord {
  public width: number = 0
  public paddingLeft: number = 0
  public paddingRight: number = 0
  public marginLeft: number = 0
  public marginRight: number = 0
  public columnGap: number = 0
  public boxSizing: string = ''

  public constructor(cssStyleDeclaration?: CSSStyleDeclaration) {
    if (cssStyleDeclaration) {
      this.width = parseInt(cssStyleDeclaration.getPropertyValue('padding-left'))
      this.paddingLeft = parseInt(cssStyleDeclaration.getPropertyValue('padding-left'))
      this.paddingRight = parseInt(cssStyleDeclaration.getPropertyValue('padding-right'))
      this.marginLeft = parseInt(cssStyleDeclaration.getPropertyValue('margin-left'))
      this.marginRight = parseInt(cssStyleDeclaration.getPropertyValue('margin-right'))
      this.columnGap = parseInt(cssStyleDeclaration.getPropertyValue('column-gap'))
      this.boxSizing = cssStyleDeclaration.getPropertyValue('box-sizing')
    }
  }
}
```

Reply:
Interesting and useful to be able to do some type casting of numbers vs strings. Since most of the time we want the number value not the px value. 

That's, a, lot, of properties though

Reply:
my vote is #2, just return `Ref<CSSStyleDeclaration | undefined>`

Reply:
Seems the simplest. 👍